File: app/src/main/java/in/hridayan/ashell/utils/HapticUtils.java
package in.hridayan.ashell.utils;

import android.view.HapticFeedbackConstants;
import android.view.View;
import in.hridayan.ashell.config.Preferences;

public class HapticUtils {

  /** Types of vibration. */
  public enum VibrationType {
    Weak,
    Strong
  }

  public static void vibrate(View view, VibrationType type) {
    if (Preferences.getHapticsAndVibration()) {
      switch (type) {
        case Weak:
          view.performHapticFeedback(HapticFeedbackConstants.CLOCK_TICK);
          break;
        case Strong:
          view.performHapticFeedback(HapticFeedbackConstants.LONG_PRESS);
          break;
      }
    }
  }

  public static void weakVibrate(View view) {
    vibrate(view, VibrationType.Weak);
  }

  public static void strongVibrate(View view) {
    vibrate(view, VibrationType.Strong);
  }
}

File: app/src/main/java/in/hridayan/ashell/utils/HapticUtils.kt
package `in`.hridayan.ashell.utils

import android.view.HapticFeedbackConstants
import android.view.View
import `in`.hridayan.ashell.config.Preferences

object HapticUtils {
    private fun vibrate(view: View?, type: VibrationType) {
        if (Preferences.getHapticsAndVibration()) {
            when (type) {
                VibrationType.Weak -> view?.performHapticFeedback(HapticFeedbackConstants.CLOCK_TICK)
                VibrationType.Strong -> view?.performHapticFeedback(HapticFeedbackConstants.LONG_PRESS)
            }
        }
    }

    @JvmStatic
    fun weakVibrate(view: View?) {
        vibrate(view, VibrationType.Weak)
    }

    fun strongVibrate(view: View?) {
        vibrate(view, VibrationType.Strong)
    }

    /** Types of vibration.  */
    enum class VibrationType {
        Weak,
        Strong
    }
}
